docs/sdks/typescript/index.md

```diff
@@ -55,927 +67,740 @@ Each SpacetimeDB client depends on some bindings specific to your module. Create
-| `name_or_address` | `string`               | The name or address of the SpacetimeDB module.                                                                                                    |
-| `auth_token?`     | `string`               | The credentials to use to connect to authenticate with SpacetimeDB.                                                                               |
-| `protocol?`       | `"binary"` \| `"json"` | Define how encode the messages: `"binary"` \| `"json"`. Binary is more efficient and compact, but JSON provides human-readable debug information. |
+Construct a `DbConnection` by calling `DbConnection.builder()` and chaining configuration methods, then calling `.build()`. You must at least specify `withUri`, to supply the URI of the SpacetimeDB to which you published your module, and `with_module_name`, to supply the human-readable SpacetimeDB domain name or the raw `Identity` which identifies the module.
 
-#### Example
+| Name                                                  | Description                                                                          |
+|-------------------------------------------------------|--------------------------------------------------------------------------------------|
+| [`withUri` method](#method-withuri)                   | Set the URI of the SpacetimeDB instance which hosts the remote database.             |
+| [`withModuleName` method](#method-withmodulename)     | Set the name or `Identity` of the remote database.                                   |
+| [`onConnect` callback](#callback-onconnect)           | Register a callback to run when the connection is successfully established.          |
+| [`onConnectError` callback](#callback-onconnecterror) | Register a callback to run if the connection is rejected or the host is unreachable. |
+| [`onDisconnect` callback](#callback-ondisconnect)     | Register a callback to run when the connection ends.                                 |
+| [`withToken` method](#method-withtoken)               | Supply a token to authenticate with the remote database.                             |
+| [`build` method](#method-build)                       | Finalize configuration and connect.                                                  |
 
-```ts
-const host = 'ws://localhost:3000';
-const name_or_address = 'database_name';
-const auth_token = undefined;
-const protocol = 'binary';
+#### Method `withUri`
 
-var spacetimeDBClient = new SpacetimeDBClient(
-  host,
-  name_or_address,
-  auth_token,
-  protocol
-);
+```typescript
+DbConnectionBuilder.withUri(uri: string): DbConnectionBuilder
 ```
 
-## Class methods
+Configure the URI of the SpacetimeDB instance or cluster which hosts the remote module.
 
-### `SpacetimeDBClient.registerReducers`
+#### Method `withModuleName`
 
-Registers reducer classes for use with a SpacetimeDBClient
+```typescript
+DbConnectionBuilder.withModuleName(name_or_identity: string): DbConnectionBuilder
```

Comment:
Probably easier to update this in a proper editor rather than adding suggestions, but ditto on all of them.